cdap-ui/app/cdap/components/DataPrep/Directives/Parse/Modals/ExcelModal.js

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, { Component, PropTypes } from 'react';
+import { Modal, ModalHeader, ModalBody, ModalFooter,FormGroup, Label, Input } from 'reactstrap';
+import isNil from 'lodash/isNil';
+import MouseTrap from 'mousetrap';
+
+import T from 'i18n-react';
+
+const PREFIX = 'features.DataPrep.Directives.Parse.Parsers.EXCEL';
+
+export default class ExcelModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      sheetSource: 'sheetnumber',
+      sheetName: null,
+      sheetNumber: 0
+    };
+    this.numberTextBox = null;
+    this.onSheetNumberChange = this.onSheetNumberChange.bind(this);
+    this.onSheetNameChange = this.onSheetNameChange.bind(this);
+    this.onSheetSourceChange = this.onSheetSourceChange.bind(this);
+    this.isApplyEnabled = this.isApplyEnabled.bind(this);
+    this.applyDirective = this.applyDirective.bind(this);
+  }
+  componentDidMount() {
+    MouseTrap.bind('enter', this.applyDirective);
+    if (this.numberTextBox) {
+      this.numberTextBox.focus();
+    }
+  }
+
+  componentWillUnmount() {
+    MouseTrap.unbind('enter');
+  }
+
+  isApplyEnabled() {
+    if (this.state.sheetSource === 'sheetnumber') {
+      return typeof this.state.sheetNumber !== 'number' || isNaN(this.state.sheetNumber);
+    }
+    if (this.state.sheetSource === 'sheetname') {
+      return isNil(this.state.sheetName) || this.state.sheetName.length === 0;
+    }
+    return false;
+  }
+  onSheetNumberChange(e) {
+    let value = parseInt(e.target.value, 10);
+    if (value < 0) {
+      return;
+    }
+    this.setState({
+      sheetNumber: parseInt(e.target.value, 10)
+    });
+  }
+  onSheetNameChange(e) {
+    this.setState({
+      sheetName: e.target.value
+    });
+  }
+  onSheetSourceChange(e) {
+    this.setState({
+      sheetSource: e.target.value
+    });
+  }
+  applyDirective() {
+    if (!this.isApplyEnabled) {
```

Comment:
Have to remove the negation here, since I was able to Enter even when the Apply button was disabled.